### PR TITLE
John conroy/ addl soft assay fields

### PR DESCRIPTION
**Pull request by john-conroy**
- Adds `soft_assaytype` to the document so we no longer have to call the ingest-api from portal-ui or portal-visualization.
- Tries to get the `pipeline` from the `pipeline-shorthand` instead of depending on regex.

### CODE REVIEW

File: src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py
Comment (NickAkhmetov) [inline src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py:144]: Could we also handle the `Image Pyramid` and `EPIC` cases in the search index? We currently have to apply some workarounds for those cases since they don't have pipeline-shorthand info: https://github.com/hubmapconsortium/portal-ui/blob/37143cf5b4af35c27f528f6ec5ae6378f30686b4/context/app/static/js/components/detailPage/DatasetRelationships/hooks.ts#L40-L58

Comment (john-conroy) [reply src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py:144]: Does it make sense to handle the EPIC case here?

Comment (john-conroy) [reply src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py:144]: The regex fallback is still in place here and image pyramids do have a `pipeline` of 'Image Pyramid' in their docs currently because of that. I can have the logic prefer 'Image Pyramid Generation' over 'Image Pyramid' here? This would update the labels in the table of contents I believe. 

https://portal.hubmapconsortium.org/browse/dataset/7e3f3b5d1890f9612efb91d3a820cdcc#section-hbm265.ljvk.224.

Comment (NickAkhmetov) [reply src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py:144]: Right; it'd definitely be ideal to keep the `pipeline` values intact. 
To clarify my ask, we currently rely on the `pipeline-shorthand` for the step details in the relationship diagram for non-image pyramids, which is blank for Image Pyramid and EPIC cases, so I'm wondering if it would make sense to populate that field in search to simplify the front-end logic, since it's otherwise blank for those dataset types.

Comment (NickAkhmetov) [reply src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py:144]: Per a slack conversation:
* `pipeline` should be the single source of truth, no need to use `pipeline-shorthand` on the frontend anymore since it's populated from that field; image pyramids will retain `image pyramid` as the description.
* EPIC `pipeline` values will be populated from the description.

**Comment by john-conroy** [comment]
@NickAkhmetov to confirm adding `soft_assaytype` we should be able to update https://github.com/hubmapconsortium/portal-visualization/blob/dbcc9aeca5e5e49b2cd44d9cdb4fabd7badd60a1/src/portal_visualization/client.py#L277 and https://github.com/hubmapconsortium/portal-visualization/blob/dbcc9aeca5e5e49b2cd44d9cdb4fabd7badd60a1/src/portal_visualization/builder_factory.py#L85 to use the doc?

**Comment by NickAkhmetov** [comment]
@john-conroy As far as I can tell, those usages should all be replaceable. 👍🏻

**Review by yuanzhou — APPROVED** [review]

**Comment by yuanzhou** [comment]
@lchoy, still waiting for you approval before I merge and deploy on DEV

**Review by lchoy — APPROVED** [review]

**Pull request merged**
